posthog/api/action.py

```diff
@@ -282,9 +281,29 @@ def _serialize_action(self, action: Action, filters: Dict[Any, Any], request: re
             'breakdown': []
         }
         if request.GET.get('shown_as', 'Volume') == 'Volume':
-            append.update(self._aggregate_by_day(action=action, filters=filters, request=request))
+            filtered_events = Event.objects.filter_by_action(action)
+            append.update(self._aggregate_by_day(filtered_events=filtered_events, filters=filters, request=request))
         elif request.GET['shown_as'] == 'Stickiness':
-            append.update(self._stickiness(action=action, filters=filters, request=request))
+            filtered_events = Event.objects.filter_by_action(action, order_by=None)
+            append.update(self._stickiness(filtered_events=filtered_events, filters=filters, request=request))
+        return append
+
+    def _serialize_event(self, event: Dict[Any, Any], request: request.Request) -> Dict:
```

Comment:
It also feels like serialize_action and serialize_event are similar enough (the only difference being `filter_by_action` or `filter`) that these could be merged into one 'serialize' object, with just that queryset being moved into `def trends`